pandas/tseries/index.py

```diff
@@ -1202,9 +1202,10 @@ def intersection(self, other):
               not other.offset.isAnchored() or
               (not self.is_monotonic or not other.is_monotonic)):
             result = Index.intersection(self, other)
-            if isinstance(result, DatetimeIndex):
-                if result.freq is None:
-                    result.offset = to_offset(result.inferred_freq)
+            result = self._simple_new(result._values, name=result.name,
```

Comment:
maybe you misunderstand, simply pass the new freq 

``result = self._shallow_copy(...., freq=....)``

We don't use ``_simple_new`` at all (outside of a few specific places), this is not one of them. This is all documented next to these functions.